src/features/calendar/utils/clusterEventsForWeekCalender.spec.ts

```diff
@@ -0,0 +1,227 @@
+import { CLUSTER_TYPE } from 'features/campaigns/hooks/useClusteredActivities';
+import clusterEventsForWeekCalender from './clusterEventsForWeekCalender';
+import { ZetkinEvent } from 'utils/types/zetkin';
+import {
+  ACTIVITIES,
+  EventActivity,
+} from 'features/campaigns/models/CampaignActivitiesModel';
+
+const mockEventData: ZetkinEvent = {
+  activity: {
+    id: 1,
+    title: 'Flyering',
+  },
+  campaign: {
+    id: 1,
+    title: 'My campaign',
+  },
+  cancelled: null,
+  contact: null,
+  end_time: '1857-07-05T13:37:00.000Z',
+  id: 1,
+  info_text: '',
+  location: {
+    id: 1,
+    lat: 0,
+    lng: 0,
+    title: 'Dorfplatz',
+  },
+  num_participants_available: 0,
+  num_participants_required: 0,
+  organization: {
+    id: 1,
+    title: 'KPD',
+  },
+  published: null,
+  start_time: '1857-07-05T13:37:00.000Z',
+};
+
+function mockEvent(id: number, data: Partial<ZetkinEvent>): EventActivity {
+  return {
+    data: {
+      ...mockEventData,
+      ...data,
+      id,
+    },
+    endDate: null,
+    kind: ACTIVITIES.EVENT,
+    startDate: null,
+  };
+}
+
+describe('doArbitraryClustering()', () => {
+  it('should not events that do not overlap', () => {
+    const result = clusterEventsForWeekCalender([
+      mockEvent(1, {
+        // 13:00 - 14:00
+        end_time: '1857-07-05T14:00:00.000Z',
+        start_time: '1857-07-05T13:00:00.000Z',
+      }),
+      mockEvent(2, {
+        // 15:00 - 17:00
+        end_time: '1857-07-05T17:00:00.000Z',
+        start_time: '1857-07-05T15:00:00.000Z',
+      }),
+    ]);
+    expect(result.length).toBe(3);
+    expect(result[0].length).toBe(2);
+    expect(result[1].length).toBe(0);
+    expect(result[2].length).toBe(0);
+  });
+
+  it('should put overlapping clusters in separate lanes', () => {
```

Comment:
One thing that's not clear to me from these tests is what the clustering is doing? Is there a way to put in here, "Groups overlapping events in to clusters"? Small thing but would be really useful to make the tests more descriptive of the code

Reply:
Just realised that clustering logic was implemented elsewhere! Lol ok disregard above.